src/uu/tail/src/args.rs

```diff
@@ -0,0 +1,501 @@
+                    ));
+                }
+            }
+        }
+
+        settings.mode = FilterMode::from(matches)?;
+
+        // Mimic GNU's tail for -[nc]0 without -f and exit immediately
+        if settings.follow.is_none()
+            && matches!(
+                settings.mode,
+                FilterMode::Lines(Signum::MinusZero, _) | FilterMode::Bytes(Signum::MinusZero)
+            )
+        {
+            std::process::exit(0)
+        }
+
+        let mut inputs: VecDeque<Input> = matches
+            .get_many::<String>(options::ARG_FILES)
+            .map(|v| v.map(|string| Input::from(string.clone())).collect())
+            .unwrap_or_default();
+
+        // apply default and add '-' to inputs if none is present
+        if inputs.is_empty() {
+            inputs.push_front(Input::default());
+        }
+
+        settings.verbose = (matches.contains_id(options::verbosity::VERBOSE) || inputs.len() > 1)
+            && !matches.contains_id(options::verbosity::QUIET);
+
+        settings.inputs = inputs;
+
+        settings.presume_input_pipe = matches.contains_id(options::PRESUME_INPUT_PIPE);
+
+        Ok(settings)
+    }
+}
+
+pub fn arg_iterate<'a>(
+    mut args: impl uucore::Args + 'a,
+) -> Result<Box<dyn Iterator<Item = OsString> + 'a>, Box<(dyn UError)>> {
+    // argv[0] is always present
+    let first = args.next().unwrap();
+    if let Some(second) = args.next() {
+        if let Some(s) = second.to_str() {
+            match parse::parse_obsolete(s) {
+                Some(Ok(iter)) => Ok(Box::new(vec![first].into_iter().chain(iter).chain(args))),
+                Some(Err(e)) => Err(UUsageError::new(
+                    1,
+                    match e {
+                        parse::ParseError::Syntax => format!("bad argument format: {}", s.quote()),
+                        parse::ParseError::Overflow => format!(
+                            "invalid argument: {} Value too large for defined datatype",
+                            s.quote()
+                        ),
+                    },
+                )),
+                None => Ok(Box::new(vec![first, second].into_iter().chain(args))),
+            }
+        } else {
+            Err(UUsageError::new(1, "bad argument encoding".to_owned()))
+        }
+    } else {
+        Ok(Box::new(vec![first].into_iter()))
+    }
+}
+
+fn parse_num(src: &str) -> Result<(u64, bool), ParseSizeError> {
+    let mut size_string = src.trim();
+    let mut starting_with = false;
+
+    if let Some(c) = size_string.chars().next() {
+        if c == '+' || c == '-' {
+            // tail: '-' is not documented (8.32 man pages)
+            size_string = &size_string[1..];
+            if c == '+' {
+                starting_with = true;
+            }
+        }
+    } else {
+        return Err(ParseSizeError::ParseFailure(src.to_string()));
+    }
+
+    parse_size(size_string).map(|n| (n, starting_with))
+}
+
+// TODO: delete ?? not used anymore
```

Comment:
Please explain this todo. Is it about the function? If it's really not used it should indeed be removed.

Reply:
Tangentially related (and should not be fixed in this PR), but it feels like the `platform` module should handle this case correctly for Windows.

Reply:
I've found that this function has no usages (anymore) within tail, so it's subject to deletion if I don't find any use for it.